ivy/functional/frontends/torch/blas_and_lapack_ops.py

```diff
@@ -193,3 +193,12 @@ def vdot(input, other, *, out=None):
         raise RuntimeError("input must be 1D vectors")
     input, other = torch_frontend.promote_types_of_torch_inputs(input, other)
     return ivy.vecdot(input, other, out=out)
+
+
+@to_ivy_arrays_and_back
+def dot(input, other, *, out=None):
+    if len(ivy.shape(input)) == 1 or len(ivy.shape(other)) == 1:
```

Comment:
I think the torch dot function requires both input and other to be 1D not just one of them. So this should be `and` not `or`. Feel free to correct me.

Reply:
Yes, you are right. Both the input and the other should be 1D.